public/src/components/trading_terminal/PlaceOrderBox.tsx

```diff
@@ -88,21 +89,30 @@ export class PlaceOrderBox extends React.Component<PlaceOrderBoxProps, {}> {
         const stockInputField = document.getElementById(orderType+"-"+orderAction+"-count") as HTMLInputElement;
         const stockCount = Number(stockInputField.value);
         const expectedCostField = document.getElementById(orderType+"-"+orderAction+"-estimation")!;
+        const orderFeeField = document.getElementById(orderType+"-"+orderAction+"-orderfee-estimation")!;
         if (isNaN(stockCount) || stockCount <= 0) {
             expectedCostField.innerHTML = "0.00";
+            orderFeeField.innerHTML = "0.00";
             return;
         }
         if (orderType == "market") {
-            expectedCostField.innerHTML = String(this.props.currentPrice * stockCount) + ".00";
+            const cost = this.props.currentPrice * stockCount;
+            const orderFee = Math.floor(this.props.orderFeePercent * cost * stockCount /100);
+            expectedCostField.innerHTML = (orderAction ==='buy')? (String(cost) + ".00") : String(stockCount) ;
+            orderFeeField.innerHTML = String(orderFee) + ".00";
```

Comment:
Using .innerHTML is bad. Not sure why we did it either. It's fine for now but just something to remember for future purposes :stuck_out_tongue: